File: auction-app-backend/src/main/java/com/internship/auctionapp/services/user/UserService.java
package com.internship.auctionapp.services.user;

import com.internship.auctionapp.models.AuthResponse;
import com.internship.auctionapp.models.LoginResponse;
import com.internship.auctionapp.models.User;
import com.internship.auctionapp.requests.UpdateUserDataRequest;
import com.internship.auctionapp.requests.UserLoginRequest;
import com.internship.auctionapp.requests.UserRegisterRequest;
import com.internship.auctionapp.requests.UserSocialLoginRequest;

import java.util.UUID;

public interface UserService {
    LoginResponse login(UserLoginRequest loginRequest);

    User register(UserRegisterRequest registerRequest);

    void logout(String token);

    AuthResponse refreshToken(String username);

    User getUser(UUID userId);

    User updateUser(UpdateUserDataRequest updateUserDataRequest, String username);

    void deactivate(String username);

    boolean exists(String email);

    LoginResponse socialLogin(UserSocialLoginRequest userSocialLoginRequest);
}

File: auction-app-ui/src/services/userService.ts
import agent from 'lib/agent';
import { User } from 'models/user';
import { UpdateUserDataRequest } from 'models/request/update/updateUserDataRequest';

const BASE_URL = '/users';

const userService = {
  deactivate: () => agent.post<any>(`${BASE_URL}/current/deactivate`, {}),
  getUser: (userId: string) => agent.get<User>(`${BASE_URL}/current/${userId}`),
  updateUser: (id: string, updateUserDataRequest: UpdateUserDataRequest) => agent.put<any>(`${BASE_URL}/${id}`, updateUserDataRequest),
  checkIfUserExists: (email: string) => agent.get<boolean>(`${BASE_URL}/exists/${email}`),
};

export default userService;
